ownership/views.py

```diff
@@ -1,194 +1,279 @@
+
+        :return:
+        """
+        delivery_type_name = self._get_request_var('delivery_type')
+        if delivery_type_name == S3DeliveryType.name:
+            return S3DeliveryType
+        else:
+            return DDSDeliveryType
+
+    def get_delivery(self):
+        transfer_id = self._get_request_var('transfer_id')
+        if transfer_id:
+            try:
+                return self.delivery_type.delivery_cls.objects.get(transfer_id=transfer_id)
+            except self.delivery_type.delivery_cls.DoesNotExist as e:
+                self.set_error_details(404, TRANSFER_ID_NOT_FOUND)
+        else:
+            self.set_error_details(400, MISSING_TRANSFER_ID_MSG)
+        return None
+
+    def set_redirect(self, target_view_name):
+        self.response_type = ResponseType.REDIRECT
+        self.redirect_target = target_view_name
+
+    def set_error_details(self, status, message):
+        self.response_type = ResponseType.ERROR
+        self.error_details = {
+            'status': status,
+            'context': {'message': message},
+        }
+
+    def set_already_complete_error(self):
+        delivery = self.delivery
+        status = State.DELIVERY_CHOICES[delivery.state][1]
+        message = "This delivery has already been processed: {}.".format(status)
+        self.set_error_details(400, message)
+
+    def make_error_response(self):
+        return render_to_response('ownership/error.html',
+                                  status=self.error_details.get('status'),
+                                  context=self.error_details.get('context'))
+
+    def _get_query_string(self):
+        query_dict = {}
+        if self.delivery:
+            query_dict['transfer_id'] = self.delivery.transfer_id
+        if self.warning_message:
+            query_dict['warning_message'] = self.warning_message
+        query_dict['delivery_type'] = self.delivery_type.name
+        return '?' + urlencode(query_dict)
+
+    def make_redirect_response(self):
+        return redirect(reverse(self.redirect_target) + self._get_query_string())
+
+    # View handlers
+    def get(self, request):
+        self._prepare(request)
+        # If preparation failed, do not handle the request
+        if not self.error_details:
+            self.handle_get()
+        return self._respond()
+
+    def post(self, request):
+        self._prepare(request)
+        # If preparation failed, do not handle the request
+        if not self.error_details:
+            self.handle_post()
+        return self._respond()
+
+
+class PromptView(DeliveryViewBase):
     """
-    Lookup url for name and append transfer_id to url.
+    Initial landing view, prompting for accept or decline button
+    Posts to ProcessDeliveryView
     """
-    url = reverse(name)
-    if transfer_id:
-        url = "{}?transfer_id={}".format(url, transfer_id)
-    return url
+    http_method_names = ['get']
+    template_name = 'ownership/index.html'
```

Comment:
I think this view needs to take into account the status of the delivery. As it is now I am prompted to ACCEPT or DECLINE on complete deliveries.

Reply:
Added in 29679ad2ae4e5670e271d1fb4c95b89d5e277a50